cmd/helm/completion.go

```diff
@@ -77,7 +78,25 @@ func runCompletion(out io.Writer, cmd *cobra.Command, args []string) error {
 }
 
 func runCompletionBash(out io.Writer, cmd *cobra.Command) error {
-	return cmd.Root().GenBashCompletion(out)
+	err := cmd.Root().GenBashCompletion(out)
+
+	// In case the user renamed the helm binary (e.g., to be able to run
+	// both helm2 and helm3), we hook the new binary name to the completion function
+	if binary := filepath.Base(os.Args[0]); binary != "helm" {
+		renamedBinaryHook := `
+# Hook the command used to generate the completion script
+# to the helm completion function to handle the case where
+# the user renamed the helm binary
+if [[ $(type -t compopt) = "builtin" ]]; then
+    complete -o default -F __start_helm ` + binary + `
```

Comment:
Consider changing this to use format notation (`%s`) and then doing a `fmt.Fprintf(out, renamedBinaryHook, binary, binary)`. It is a bit safer than trying to glue the various string components together

Reply:
Better.  Thanks.